agent/src/main/scala/za/co/absa/atum/agent/dispatcher/HttpDispatcher.scala

```diff
@@ -61,13 +59,13 @@ class HttpDispatcher(config: Config) extends Dispatcher with Logging {
 
     val response = backend.send(request)
 
-    safeResponseBody(response).get
+    safeResponseBody(response)
   }
 
-  private def safeResponseBody(response: Response[Either[String, String]]): Try[String] = {
+  private def safeResponseBody(response: Response[Either[String, String]]): String = {
```

Comment:
Absolutely, I like the change, going from Either to Try and then calling .get is an anti-pattern. Throwing exceptions can sometimes be disruptive and it's often better to let the client code decide how to handle error cases. I wonder if we could instead return Either[HttpException, String]. This way we would be using type system to explicitly signal that the operation might fail with an HttpException. We could also have two flavours of saveCheckpoint method (one throwing the exception, another returning Either). This would give the client code flexibility to choose the version that best fits their error handling strategy.

Reply:
Btw I am not sure about the method's name. Can you think of sth better?

Reply:
changed